plc4j/utils/opm/src/main/java/org/apache/plc4x/java/opm/PlcEntityManager.java

```diff
@@ -0,0 +1,380 @@
+     * @param <T>   type of param {@code clazz}.
+     * @return a connected entity.
+     * @throws OPMException when proxy can't be build.
+     */
+    public <T> T connect(Class<T> clazz) throws OPMException {
+        getPlcEntityAndCheckPreconditions(clazz);
+        try {
+            // Use Byte Buddy to generate a subclassed proxy that delegates all PlcField Methods
+            // to the intercept method
+            return new ByteBuddy()
+                .subclass(clazz)
+                .method(any()).intercept(MethodDelegation.to(this))
+                .make()
+                .load(Thread.currentThread().getContextClassLoader())
+                .getLoaded()
+                .getConstructor()
+                .newInstance();
+        } catch (NoSuchMethodException | InvocationTargetException | InstantiationException | IllegalAccessException e) {
+            throw new OPMException("Unable to instantiate Proxy", e);
+        }
+    }
+
+    private <T> PlcEntity getPlcEntityAndCheckPreconditions(Class<T> clazz) {
+        PlcEntity annotation = clazz.getAnnotation(PlcEntity.class);
+        if (annotation == null) {
+            throw new IllegalArgumentException("Given Class is no Plc Entity, i.e., not annotated with @PlcEntity");
+        }
+        // Check if default constructor exists
+        try {
+            clazz.getConstructor();
+        } catch (NoSuchMethodException e) {
+            throw new IllegalArgumentException("Cannot use PlcEntity without default constructor", e);
+        }
+        return annotation;
+    }
+
+    //------------------------------------------------------------------------------------------------
+    //
+    //  Methods for interception for the proxy object
+    //
+    //------------------------------------------------------------------------------------------------
+
+    /**
+     * Basic Intersector for all methods on the proxy object.
+     * It checks if the invoked method is a getter and if so, only retrieves the requested field, forwarding to
+     * the {@link #fetchValueForGetter(Object, Method)} method.
+     * <p>
+     * If the field is no getter, then all fields are refreshed by calling {@link #refetchAllFields(Object)}
+     * and then, the method is invoked.
+     *
+     * @param o    Object to intercept
+     * @param m    Method that was intercepted
+     * @param c    Callable to call the method after fetching the values
+     * @param that Reference to the proxy object
+     * @return possible result of the original methods invocation
+     * @throws OPMException Problems with plc / proxying
+     */
+    @RuntimeType
+    // TODO: avoid using single letter params.
+    public Object intercept(@This Object o, @Origin Method m, @SuperCall Callable<?> c, @Super Object that) throws OPMException {
+        LOGGER.trace("Invoked method {} on connected PlcEntity {}", m.getName(), that);
+
+        if (m.getName().startsWith("get")) {
```

Comment:
maybe we should assert that no method starting with get and having attributes get picked up

Reply:
This is a fair point. And in fact, we should also support `isXXX()` for boolean fields.
We should add a Todo here, or better extract this in a method `isGetter(m)` where we can adopt all these conditions.